controllers/operator/suite_test.go

```diff
@@ -112,6 +132,62 @@ func startTestEnv() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 	//+kubebuilder:scaffold:scheme
+
+	// Step 9: Apply the resources from the buffer to the test environment
+	klog.Info("Applying paches to the test environment")
+	err = applyResources(buffy)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func applyResources(buffer *bytes.Buffer) error {
+	// Create a universal decoder for deserializing the resources
+	decoder := scheme.Codecs.UniversalDeserializer()
+
+	// Split the buffer into individual YAML documents
+	docs := bytes.Split(buffer.Bytes(), []byte("\n---[ ]*\n"))
+	for _, doc := range docs {
+		if len(doc) == 0 {
+			continue
+		}
+
+		raw := buffer.Bytes()
+		if len(raw) == 0 {
+			return nil // Nothing to process
+		}
+
+		// Decode the resource from the buffer
+		obj, _, err := decoder.Decode(raw, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		// Check if the resource already exists
+		existingObj := obj.DeepCopyObject().(client.Object)
+		err = k8sClient.Get(context.TODO(), client.ObjectKey{
+			Name:      existingObj.GetName(),
+			Namespace: existingObj.GetNamespace(),
+		}, existingObj)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Resource does not exist, create it
+				err = k8sClient.Create(context.TODO(), obj.(client.Object))
+				if err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		} else {
+			// Resource exists, update it
+			obj.(client.Object).SetResourceVersion(existingObj.GetResourceVersion())
+			err = k8sClient.Update(context.TODO(), obj.(client.Object))
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
```suggestion
		if err != nil && !errors.IsNotFound(err) {
			// Return error if it's not a "not found" error
			return err
		}
		if err == nil {
			// Resource exists, update it
			obj.(client.Object).SetResourceVersion(existingObj.GetResourceVersion())
			err = k8sClient.Update(ctx, obj.(client.Object))
			if err != nil {
				return err
			}
		} else { // errors.IsNotFound(err) is true
			// Resource does not exist, create it
			err = k8sClient.Create(ctx, obj.(client.Object))
			if err != nil {
				return err
			}
		}
```

More about taste here: to avoid many nested if statements.

Reply:
Also, consider if we really need this code now (we should always hit the Create branch, right?)

Reply:
Ill test it

Reply:
I dont think I can remove the update resource segment. When I run `make test` without it I get the following error

```
[BeforeSuite] 
/home/user/Working/multiarch-manager-operator/controllers/operator/suite_test.go:79
  Enter [BeforeSuite] TOP-LEVEL - /home/user/Working/multiarch-manager-operator/controllers/operator/suite_test.go:79 @ 06/07/24 15:26:57.682
  STEP: bootstrapping test environment - /home/user/Working/multiarch-manager-operator/controllers/operator/suite_test.go:99 @ 06/07/24 15:26:57.683
  2024-06-07T15:26:57-07:00	DEBUG	controller-runtime.test-env	starting control plane
  2024-06-07T15:27:00-07:00	DEBUG	controller-runtime.test-env	installing CRDs
  2024-06-07T15:27:00-07:00	DEBUG	controller-runtime.test-env	reading CRDs from path	{"path": "../../config/crd/bases"}
  2024-06-07T15:27:00-07:00	DEBUG	controller-runtime.test-env	read CRDs from file	{"file": "multiarch.openshift.io_clusterpodplacementconfigs.yaml"}
  2024-06-07T15:27:00-07:00	DEBUG	controller-runtime.test-env	installing CRD	{"crd": "clusterpodplacementconfigs.multiarch.openshift.io"}
  2024-06-07T15:27:00-07:00	DEBUG	controller-runtime.test-env	adding API in waitlist	{"GV": "multiarch.openshift.io/v1alpha1"}
  2024-06-07T15:27:00-07:00	DEBUG	controller-runtime.test-env	installing webhooks
I0607 15:27:00.543666 1966467 suite_test.go:124] Applying CRDs to the test environment
  [FAILED] Unexpected error:
      <*errors.StatusError | 0xc0007c0280>: 
      customresourcedefinitions.apiextensions.k8s.io "clusterpodplacementconfigs.multiarch.openshift.io" already exists
      {
        ...
              Status: "Failure",
              Message: "customresourcedefinitions.apiextensions.k8s.io \"clusterpodplacementconfigs.multiarch.openshift.io\" already exists",
              Reason: "AlreadyExists",
              ...
  Full Stack Trace
    github.com/openshift/multiarch-tuning-operator/controllers/operator.startTestEnv()
    	/home/user/Working/multiarch-manager-operator/controllers/operator/suite_test.go:129 +0xd54
    github.com/openshift/multiarch-tuning-operator/controllers/operator.glob..func2()
    	/home/user/Working/multiarch-manager-operator/controllers/operator/suite_test.go:85 +0x2ae
```